core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -571,6 +573,14 @@ class TaskManager(taskManagerResources: SystemResources = TaskManagerDefaults.de
     logger.debug("stepExecution: finishing one round of execution")
     logger.debug("stepExecution: found " + runningTasks.size + " running tasks and " + tasksToSchedule.size + " tasks to schedule")
 
+    // Update the current sleep time: exponential reduction if we could do **anything**, otherwise linear increase.
+    if (canDoAnything) {
+      curSleepMilliseconds = curSleepMilliseconds / 2
+    }
+    else {
+      curSleepMilliseconds = Math.min(sleepMilliseconds, curSleepMilliseconds + (sleepMilliseconds / 10))
```

Comment:
I think this is a little sketchy since `sleepMilliseconds` is still a parameter.  What if someone set that to 5ms?  Then you'll never increase past 5ms because `sleepMilliseconds / 10` is 0.  I'll echo my earlier comment that we should remove the constructor parameter, and have constants for `initialSleepMillis`, `minSleepMillis` and `maxSleepMillis` and then use the strategy you have for adjustment.